File: Ambulance.BLL/Models/AnaliticsModels/FullAnalyticsModel.cs
namespace Ambulance.BLL.Models;

public class FullAnalyticsModel
{
    public CallAnalyticsModel Calls { get; set; } = new();
    public List<BrigadeResourceModel> Brigades { get; set; } = new();
    public List<DecAllergAnalyticsModel> Deceases { get; set; } = new();
}

File: ambulance-site/src/api/models/AnaliticsModels/FullAnalyticsModel.js
import { CallAnalyticsModel } from "./CallAnalyticsModel.js";
import { BrigadeResourceModel } from "./BrigadeResourceModel.js";
import { DecAllergAnalyticsModel } from "./DecAllergAnalyticsModel.js";

export class FullAnalyticsModel {
    constructor({ calls = {}, brigades = [], deceases = [] } = {}) {
        this.calls = new CallAnalyticsModel(calls);
        this.brigades = brigades.map(b => new BrigadeResourceModel(b));
        this.deceases = deceases.map(d => new DecAllergAnalyticsModel(d));
    }
}
